generators/server/cleanup.js

```diff
@@ -9,154 +9,141 @@
-      }
-      if (generator.isJhipsterVersionLessThan('6.5.2')) {
-        generator.removeFile(`${testDir}service/mapper/UserMapperIT.java`);
-        generator.removeFile(`${javaDir}service/${generator.upperFirstCamelCase(generator.baseName)}KafkaConsumer.java`);
-        generator.removeFile(`${javaDir}service/${generator.upperFirstCamelCase(generator.baseName)}KafkaProducer.java`);
-        generator.removeFile(`${testDir}web/rest/ClientForwardControllerIT.java`);
-      }
-      if (generator.isJhipsterVersionLessThan('6.6.1')) {
-        generator.removeFile(`${javaDir}web/rest/errors/EmailNotFoundException.java`);
-        generator.removeFile(`${javaDir}config/DefaultProfileUtil.java`);
-        generator.removeFolder(`${javaDir}service/util`);
-      }
-      if (generator.isJhipsterVersionLessThan('6.8.0')) {
-        generator.removeFile(`${javaDir}security/oauth2/JwtAuthorityExtractor.java`);
-      }
-      if (generator.isJhipsterVersionLessThan('6.8.1')) {
-        generator.removeFile(`${javaDir}config/ReactivePageableHandlerMethodArgumentResolver.java`);
-        generator.removeFile(`${javaDir}config/ReactiveSortHandlerMethodArgumentResolver.java`);
-      }
-      if (generator.isJhipsterVersionLessThan('7.0.0-beta.0')) {
-        generator.removeFile(`${javaDir}config/apidoc/SwaggerConfiguration.java`);
-        generator.removeFile(`${javaDir}config/audit/package-info.java`);
-        generator.removeFile(`${javaDir}config/audit/AuditEventConverter.java`);
-        generator.removeFile(`${javaDir}domain/PersistentAuditEvent.java`);
-        generator.removeFile(`${javaDir}repository/PersistenceAuditEventRepository.java`);
-        generator.removeFile(`${javaDir}repository/CustomAuditEventRepository.java`);
-        generator.removeFile(`${javaDir}service/AuditEventService.java`);
-        generator.removeFile(`${javaDir}web/rest/AuditResource.java`);
-        generator.removeFile(`${testDir}service/AuditEventServiceIT.java`);
-        generator.removeFile(`${testDir}web/rest/AuditResourceIT.java`);
-        generator.removeFile(`${testDir}repository/CustomAuditEventRepositoryIT.java`);
+  if (generator.isJhipsterVersionLessThan('3.5.0')) {
+    generator.removeFile(`${javaDir}domain/util/JSR310DateTimeSerializer.java`);
+    generator.removeFile(`${javaDir}domain/util/JSR310LocalDateDeserializer.java`);
+  }
+  if (generator.isJhipsterVersionLessThan('3.6.0')) {
+    generator.removeFile(`${javaDir}config/HerokuDatabaseConfiguration.java`);
+  }
+  if (generator.isJhipsterVersionLessThan('3.10.0')) {
+    generator.removeFile(`${javaDir}config/CloudMongoDbConfiguration.java`);
+    generator.removeFile(`${javaDir}security/CustomAccessDeniedHandler.java`);
+    generator.removeFile(`${javaDir}web/filter/CsrfCookieGeneratorFilter.java`);
+  }
+  if (generator.isJhipsterVersionLessThan('3.12.0')) {
+    generator.removeFile(`${javaDir}config/hazelcast/HazelcastCacheRegionFactory.java`);
```

Comment:
Could be nice to have multiple functions for each JH option: this one for example could test in the cacheManagerHazelcast has been activated.

It can help for generator componentization in the mid term WDYT?